articles/working-with-lambda-expressions-and-anonymous-functions/index.md

```diff
@@ -0,0 +1,114 @@
+
+Lambda expressions can be challenging to understand. Types that look like this: `(Int) -> String` are not intuitive to many developers. This article aims to simplify the concept behind lambda expressions and how to use thm.
+
+### Prerequisites
+
+1. A basic understanding of the Kotlin programming language.
+2. Either IntelliJ IDEA or access to [Kotlin Playground](https://play.kotlinlang.org/)
+
+### Table of Contents
+
+1. What is a lambda expression.
+2. A use case of lambda expressions.
+3. Summary.
+
+### What is a Lambda Expression?
+
+A lambda expression is a function that you pass only as an expression. Lambda expressions have the following structure: `{}`. They are a specialized language construct that helps represent functions in Kotlin.
+
+Unlike in Java, functions in Kotlin are first class. This means that you can store them as variables and in data structures. You can pass them as arguments and can also be a return value.
+
+Since they can you can store them as variables, they must have a type. Kotlin uses function types to represent them. Function types have the following structure:
+`(parameterType) -> returnType`
+
+Variables that are function types expect a lambda expression as demonstrated below:
+
+- `val a:Int` expects a value such as `1`,`2` or `3`.
+
+- `val b:String` expects a value such as `"string"` or `"name"`.
+
+- `val c:Boolean` expects a value such as `true` or `false`.
+
+In that spirit:
+
+- `val d:() -> Int` expects a value such as `{1}` or `{2}`
+
+- `val e:() -> String` expects a value such as `{"string"}` or `{"name"}`
+
+When we introduce parameters:
+
+`val f:(Int) -> String` expects a value such as `{"string"}` or `{"name"}`
+
+Note that you don't have to use the value passed in the parameter. But you have to respect the return type and return the appropriate value.
+
+Higher-order functions are functions that can have lambda expressions passed into them. They can also return lambda expressions.
+You can find these definitions in the [official docs](https://kotlinlang.org/docs/lambdas.html)
+
+When working with all functions, there are 3 main steps to follow:
+
+**Step 1. Creating the function:** this involves giving the function its name, parameters, and return type.
+
+**Step 2. Implementing the function:** this is where you define the task that the function should execute.
+
+**Step 3. Calling the function:** here you call the function and give it the parameters it requires to perform its task.
+
+> **Note**: This article refers to functions in the context of functional programming. In object-oriented programming, you can use interfaces and abstract classes. They can postpone function implementation. Later, you can override the function when you implement the interface or inherit the class.
+
+### A use case of lambda expressions
+
+Regular functions are created in the same place where they implement their functionality. Have a look at this function:
+
+```kotlin
+fun add(a:Int, b:Int){
```

Comment:
Give your functions a short name to make them more readable and the code to be cleaner. A technical suggestion here.